ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -80,16 +82,20 @@
   private final StoreKeyFactory factory;
   private final File bloomFile;
   private final StoreMetrics metrics;
-  // an approximation of the last modified time.
-  private final AtomicLong lastModifiedTimeSec;
   private final AtomicInteger numberOfItems;
+  private final Time time;
 
+  // an approximation of the last modified time.
+  private AtomicLong lastModifiedTimeSec;
```

Comment:
Might give a wrong notion that value is initialized with wrong value earlier during initialization and fixed later to right value. Why do you think final is very much needed